cmd/run.go

```diff
@@ -127,11 +128,34 @@ func doRun(flags *cobra.Command, args []string) error {
 		return fmt.Errorf("unable to initialize VM: %w", err)
 	}
 
+	isRunning, err := bootcVM.IsRunning()
+	if err != nil {
+		return fmt.Errorf("unable to check if VM is running: %w", err)
+	}
+	if isRunning {
+		return fmt.Errorf("VM already running, use the ssh command to connect to it")
+	}
+
+	// create the disk image
+	bootcDisk := bootc.NewBootcDisk(containerImage, ctx, user, cache)
+	err = bootcDisk.Install(vmConfig.Quiet, diskImageConfigInstance)
+
```

Comment:
cache.Create() takes an exclusive lock and frees it after creating the cache directory. vm.NewVM() in run takes a shared lock in main, so this is not introducing a new bug. This will be much simpler to fix with the cache/config loading refactor I am working on for a separate MR. I'd rather not try to fix everything in this MR and just keep the scope to what is needed for the disk size fix.